sdks/go/pkg/beam/util/gcsx/gcs.go

```diff
@@ -39,6 +41,15 @@ func NewClient(ctx context.Context, scope string) (*storage.Service, error) {
 	return storage.New(cl)
```

Comment:
Is there any way to make the client load the credentials internally on the first 401?

It seems like we should only get the credentials if they are asked. We should still be able to pass in the scope that we want for the client.

Reply:
I see your point, but not a good way to do that. It seems to go against the flow of these libraries and we'd have to do far more work manually to achieve that. I'd prefer not to take that on now. Filed https://issues.apache.org/jira/browse/BEAM-3896.